Freebsd

bcoca opened the pull request:
Commits in this request:

1)Updated makefile and tests to work with freebsd

2)Generalized ansible_python_interpreter to work with any interpreter used by modules *updated per review

3)Freebsd ports package

bcoca committed:
- Makefile now works with freebsd (date command options are diff)
date still issues warning and ignores TZ

- Updated tests to work inside bsd jails (127 addresses are an issue)
Signed-off-by: Brian Coca <briancoca+ansible@gmail.com>

bcoca committed:
Expanded from a single python interpreter variable to <any> intepreter for module shebang subsitution
based on host variables
Signed-off-by: Brian Coca <briancoca+ansible@gmail.com>

bcoca committed:
now only appends to interpreters list if it actually matches
Signed-off-by: Brian Coca <briancoca+ansible@gmail.com>

bcoca committed:
added freebsd ports package
Signed-off-by: Brian Coca <briancoca+ansible@gmail.com>

mpdehaan commented:
I'm getting numerous errors when testing this, from "make tests"

re was not imported in the file, also

if interpreters.count() > 0 is a type error

Did you mean len(interpreters) ?

Feel free to resubmit this one 'make tests' passes.

bcoca commented:
Sorry about that, I expected to make the pull request on a commit not a
branch, so I had added the untested interpreter update. After I updated the
branch and fixed those issues, I was going to reissue the request github
showed the pull request updated, from now on I'll make 'pure' feature
branches.

I'll send a new pull request, just for the BSD stuff. The interpreter
updates will be on a followup.
